src/nvim/globals.h

```diff
@@ -92,6 +92,10 @@
 # define SYNTAX_FNAME   "$VIMRUNTIME" _PATHSEPSTR "syntax" _PATHSEPSTR "%s.vim"
 #endif
 
+#ifndef SYNTAX_FILE
+# define SYNTAX_FILE   "$VIMRUNTIME" _PATHSEPSTR "syntax" _PATHSEPSTR "syntax.vim"
```

Comment:
If you're not going to reuse SYNTAX_FNAME, which is nearly identical to the purpose of this, might as well hard-code this string directly in your `do_source` call in main.c. Because this isn't going to be used anywhere else--`SYNTAX_FNAME` already exists for that purpose.